common/djangoapps/student/roles.py

```diff
@@ -153,7 +153,8 @@ def add_users(self, *users):
         # legit get updated.
         from student.models import CourseAccessRole
         for user in users:
-            if user.is_authenticated and user.is_active and not self.has_user(user):
+            user = User.objects.get(id=user.id)
```

Comment:
When we're assigning a role, the User object that we get from `ccx.coach` isn't adequate for use in database operations involving other objects. This is because within a migration, we have to use `apps.get_model`, rather than the standard import process. This method returns objects based on the historical model that inherit from a different base class than what's in the database, so we get an error stating that we can only attach/create/filter based on "User instances", rather than "User objects":

`ValueError: Cannot assign "<User: User object>": "CourseAccessRole.user" must be a "User" instance.`

This also explains why `.is_authenticated()` doesn't work; the historical, dynamically-created class doesn't carry any of the current version's methods.

Actually, I just did some further investigation, and with this particular patch, we no longer need to fake `is_authenticated`, because the rest of the process is getting a database-standard `User` object. If it's all right, I'll remove those shims, and add comments around the patches in `roles.py` explaining why they're there.